Adds release announcement blog post for the CLI for Microsoft 365 v6.3 (#1071)
pnp/blog: content/cli-for-microsoft-365/cli-for-microsoft-365-v6-3/index.md

categories:
  - CLI for Microsoft 365
images:
  - images/banner-cli-m365.png
tags:
  - CLI for Microsoft 365
  - SharePoint
  - SharePoint Framework (SPFx)
type: popular
---

We've just published a new minor version of the CLI for Microsoft 365 with new commands for working with and managing Microsoft 365 and SharePoint Framework projects on any platform.

## Manage Microsoft 365 and SharePoint Framework projects on any platform

CLI for Microsoft 365 is a cross-platform CLI that allows you to manage various configuration settings of Microsoft 365 and SharePoint Framework projects no matter which operating system or shell you use.

While building solutions for Microsoft 365 expands beyond the Windows operating system, managing many platform settings is possible only through PowerShell on Windows. As more and more users work on non-Windows machines, it's inconvenient for them to use a Windows virtual machine to configure their tenants. With the CLI for Microsoft 365, you can configure your tenant no matter which operating system you use. Additionally, using CLI for Microsoft 365, you can manage your SharePoint Framework projects.

## New version of CLI for Microsoft 365 – v6.3

> Following our monthly release cadence, we've released a new version of the CLI for Microsoft 365 with some new capabilities. The month of February was no exception. We had a ton of improvements and new commands. Here are a few of the most noteworthy additions.

Reviewer: It might be fun to mention here how many new commands were added in the shortest month of the year 😄

> For the complete list of what's new and changed, see the [release notes](https://pnp.github.io/cli-microsoft365/about/release-notes/#v630).

## Added new Azure AD commands

Working with Azure AD is one of the things for which the CLI for Microsoft 365 is often used. In this release we've introduced a series of new commands for managing users.

To create a new user, run:

```sh
m365 aad user add --displayName "John Doe" --userName "john.doe@contoso.com" --password "SomePassw0rd" --forceChangePasswordNextSignIn
```

To delete a user, run:

```sh
m365 aad user remove --name john.doe@contoso.com
```

To list license details of a specific user, run:

```sh
m365 aad user license list --userName john.doe@contoso.com
```

> Here's a list of all new Azure AD commands for this release:

Reviewer 2: Let's say we're just half way there, more aad user commands are coming 😊

Epic: https://github.com/pnp/cli-microsoft365/issues/4496

Author: I've added it to the whats next section

- [aad user add](https://pnp.github.io/cli-microsoft365/cmd/aad/user/user-add)
- [aad user remove](https://pnp.github.io/cli-microsoft365/cmd/aad/user/user-remove)
- [aad user license list](https://pnp.github.io/cli-microsoft365/cmd/aad/user/user-license-list)
- [aad user recyclebinitem clear](https://pnp.github.io/cli-microsoft365/cmd/aad/user/user-recyclebinitem-clear)
- [aad user recyclebinitem list](https://pnp.github.io/cli-microsoft365/cmd/aad/user/user-recyclebinitem-list)
- [aad user recyclebinitem remove](https://pnp.github.io/cli-microsoft365/cmd/aad/user/user-recyclebinitem-remove)

## Added new Planner Commands

> In this release another series of Microsoft Planner commands where introduced. With these commands you can manage Planner Roster, which is a new type of container that can be used to manage access to planner plans.

Reviewer 2: ```suggestion
In this release, another series of Microsoft Planner commands are introduced. With these commands you can manage Planner Roster, which is a new type of container that can be used to manage access to planner plans without the need of an Microsoft365 group.
```

To create a new Microsoft Planner Roster, run: 

```sh
m365 planner roster add
```

To give a user access to a Microsoft Planner Roster, run: 

```sh
m365 planner roster member add --rosterId tYqYlNd6eECmsNhN_fcq85cAGAnd --userName john.doe@contoso.com
```

To list all users having access to a Microsoft Planner Roster, run: 

```sh
m365 planner roster member list --rosterId tYqYlNd6eECmsNhN_fcq85cAGAnd
```

Here's a list of all new Planner commands for this release:

- [planner roster add](https://pnp.github.io/cli-microsoft365/cmd/planner/roster/roster-add)
- [planner roster get](https://pnp.github.io/cli-microsoft365/cmd/planner/roster/roster-get)
- [planner roster remove](https://pnp.github.io/cli-microsoft365/cmd/planner/roster/roster-remove)
- [planner roster member add](https://pnp.github.io/cli-microsoft365/cmd/planner/roster/roster-member-add)
- [planner roster member list](https://pnp.github.io/cli-microsoft365/cmd/planner/roster/roster-member-list)
- [planner roster member remove](https://pnp.github.io/cli-microsoft365/cmd/planner/roster/roster-member-remove)

## Added new Compliance Commands

> The last few releases introduced a lot of functionality around Microsoft 365 Purview retention labels. This release expands on that by adding new commands around event-based retention labels. Aside from data retention, a new command was introduced for accessing the audit log.

Reviewer 2: ```suggestion
The last few releases introduced a lot of functionalities around Microsoft 365 Purview retention labels. This release expands on that by adding new commands around event-based retention labels. Aside from data retention, a new command was introduced for accessing the audit log within your tenant.
```

- [purview auditlog list](https://pnp.github.io/cli-microsoft365/cmd/purview/auditlog/auditlog-list)
- [purview retentionevent get](https://pnp.github.io/cli-microsoft365/cmd/purview/retentionevent/retentionevent-get)
- [purview retentionevent list](https://pnp.github.io/cli-microsoft365/cmd/purview/retentionevent/retentionevent-list)
- [purview retentionevent remove](https://pnp.github.io/cli-microsoft365/cmd/purview/retentionevent/retentionevent-remove)
- [purview retentioneventtype add](https://pnp.github.io/cli-microsoft365/cmd/purview/retentioneventtype/retentioneventtype-add)
- [purview retentioneventtype get](https://pnp.github.io/cli-microsoft365/cmd/purview/retentioneventtype/retentioneventtype-get)
- [purview retentioneventtype list](https://pnp.github.io/cli-microsoft365/cmd/purview/retentioneventtype/retentioneventtype-list)
- [purview retentioneventtype remove](https://pnp.github.io/cli-microsoft365/cmd/purview/retentioneventtype/retentioneventtype-remove)
- [purview retentioneventtype set](https://pnp.github.io/cli-microsoft365/cmd/purview/retentioneventtype/retentioneventtype-set)

## Added new SharePoint commands

> SharePoint has always been a part of Microsoft 365 that's covered by the most commands in the CLI for Microsoft 365. In this release more commands where introduced that make managing SharePoint easier. The main subjects Application Customizers, Sharing Links and Navigation:

Reviewer 2: ```suggestion
SharePoint has always been a part of Microsoft 365 that's covered by the most commands in the CLI for Microsoft 365. In this release more commands were introduced that make managing SharePoint easier. The main subjects Application Customizers, Sharing Links and Navigation:
```

> SharePoint has always been a part of Microsoft 365 that's covered by the most commands in the CLI for Microsoft 365. In this release more commands where introduced that make managing SharePoint easier. The main subjects Application Customizers, Sharing Links and Navigation:

Author: ah, the oxford comma, @Jwaegebaert. That one is optional I presume 😁

To add an application customizer to a SharePoint site, run:

```sh
m365 spo applicationcustomizer add --title 'Some customizer' --clientSideComponentId 799883f5-7962-4384-a10a-105adaec6ffc --webUrl https://contoso.sharepoint.com/sites/sales
```

To remove all sharing links from a file, run:

```sh

> m365 spo file sharinglink clear --webUrl https://contoso.sharepoint.com/sites/demo --fileId daebb04b-a773-4baa-b1d1-3625418e3234

Reviewer 2: Maybe let's specify an example with a server relative URL which is more common for users to use.

```

To update the title of a navigation node, run:

```sh

> m365 spo navigation node set --webUrl https://contoso.sharepoint.com/sites/marketing --id 2209 --title "Pictures"

Reviewer 2: Maybe let's also add the audience targeting option here, this option was added to the existing navigation commands as well.

- [spo applicationcustomizer add](https://pnp.github.io/cli-microsoft365/cmd/spo/applicationcustomizer/applicationcustomizer-add)
- [spo file sharinglink clear](https://pnp.github.io/cli-microsoft365/cmd/spo/file/file-sharinglink-clear)
- [spo file sharinglink set](https://pnp.github.io/cli-microsoft365/cmd/spo/file/file-sharinglink-set)
- [spo navigation node get](https://pnp.github.io/cli-microsoft365/cmd/spo/navigation/navigation-node-get)
- [spo navigation node set](https://pnp.github.io/cli-microsoft365/cmd/spo/navigation/navigation-node-set)

## Making SharePoint Framework Development easier

Granting permissions in SharePoint Framework projects can be a bother. Developers need to deploy the package before the permissions are visible on the API access page of the SharePoint admin dashboard. This is experienced by many as suboptimal. In this release we introduce a new command to make the life of SPFx developers easier. You can now grant consent to all API permissions that are included in the project. No need to execute multiple gulp commands and leave VS Code. 

> You can now simply run:

Author: It's already in there. I highlighted the last few words a bit more: 'No need to leave VS Code.'

```sh 
m365 spfx project permissions grant
```

## Another steps in implementing CLI for Microsoft 365 context

> In our previous releases, we have taken the first steps to create a CLI context. This allows you to save time by storing default values for options that you often use. For example, if you're working with a specific site collection, you can save the URL of the site collection in the context, and CLI for Microsoft 365 will automatically use it when you run a command that requires the URL of the site collection. This way, you don't have to specify the URL of the site collection every time you run a command. In this release, we created an extra commands to work with the CLI context. Note this is still a work in progress. We allow managing context settings but don't apply them to commands just yet.

Reviewer 2: ```suggestion
In our previous releases, we have taken the first steps to create a CLI context. This allows you to save time by storing default values for options that you often use. For example, if you're working with a specific site collection, you can save the URL of the site collection in the context, and CLI for Microsoft 365 will automatically use it when you run a command that requires the URL of the site collection. This way, you don't have to specify the URL of the site collection every time you run a command. In this release, we created an extra command to work with the CLI context. Note this is still a work in progress. We allow managing context settings but don't apply them to commands just yet.
```

To list all options added to the context, run:

```sh
m365 context option list
```

> 

Reviewer 2: I would also make a small topic about a very much used command: `flow run list` is now capable of only retrieving flow runs of a specific status and/or within a specific time frame:

https://github.com/pnp/cli-microsoft365/issues/4388
https://pnp.github.io/cli-microsoft365/cmd/flow/run/run-list/

> ## What's next

Reviewer 2: Still half way the epic to manage aad users, so maybe let's mention this as well.

Here are some things that we are currently working on.

### Microsoft 365 Purview

We're currently busy implementing the last of a series of commands around retention labels. Aside from that, work is also being done to build new commands to manage threat assessments and sensitivity labels. 

### Manage SPFx extensions

In the upcoming weeks, we'll keep working to create more commands to add, set, list, and remove SPFx extensions such as ListView Command Sets or Application Customizers.

## Contributors

This release wouldn't be possible without the help of (in alphabetical order):

- [Adam Wójcik](https://github.com/Adam-it)
- [Jasey Waegebaert](https://github.com/Jwaegebaert)
- [Martin Lingstuyl](https://github.com/martinlingstuyl)
- [Mathijs Verbeeck](https://github.com/MathijsVerbeeck)
- [Milan Holemans](https://github.com/milanholemans)
- [Nanddeep Nachan](https://github.com/nanddeepn)
- [Nico De Cleyre](https://github.com/nicodecleyre)

> - [Nico De Cleyre](https://github.com/rabwill)

Reviewer 2: Wrong name 😊

- [Patrick Lamber](https://github.com/plamber)
- [Waldek Mastykarz](https://github.com/waldekmastykarz)

> 

Reviewer: One additional name is missing here

`- [Reshmee Auckloo](https://github.com/reshmee011)`
Contributed to a new sample script. https://github.com/pnp/cli-microsoft365/pull/4506

Author: pfew, thanks @Jwaegebaert 
Not sure how I missed that.

Thank you all for the time you chose to spend on CLI for Microsoft 365 and for your help to advance it!

### High fives

CLI for Microsoft 365 wouldn't be where it is today if it weren't for our users who provide us with feedback. High fives to the following people who took the time to share their feedback and ideas for improvement with us (in alphabetical order):

- [FinderFees](https://github.com/FinderFees)

> 

Reviewer 2: Are we mentioning people that gave feedback in February, or the people for which their feedback was implemented in February?

If  it is the first one, we can mention few more people: 

- https://github.com/pnp/cli-microsoft365/issues/4558
- https://github.com/pnp/cli-microsoft365/issues/4507
- https://github.com/pnp/cli-microsoft365/issues/4480
- https://github.com/pnp/cli-microsoft365/issues/4472
- https://github.com/pnp/cli-microsoft365/issues/4461

Also, I think that FinderFees was already mentioned in previous blog post.

Reviewer: I would go for the former 😄

> For the complete list of what's new and changed, see the [release notes](https://pnp.github.io/cli-microsoft365/about/release-notes/#v630).

## Added new Azure AD commands

Working with Azure AD is one of the things for which the CLI for Microsoft 365 is often used. In this release we've introduced a series of new commands for managing users.

To create a new user, run:

```sh
m365 aad user add --displayName "John Doe" --userName "john.doe@contoso.com" --password "SomePassw0rd" --forceChangePasswordNextSignIn
```

To delete a user, run:

```sh
m365 aad user remove --name john.doe@contoso.com
```

To list license details of a specific user, run:

```sh
m365 aad user license list --userName john.doe@contoso.com
```

Here's a list of all new Azure AD commands for this release:

- [aad user add](https://pnp.github.io/cli-microsoft365/cmd/aad/user/user-add)
- [aad user remove](https://pnp.github.io/cli-microsoft365/cmd/aad/user/user-remove)
- [aad user license list](https://pnp.github.io/cli-microsoft365/cmd/aad/user/user-license-list)
- [aad user recyclebinitem clear](https://pnp.github.io/cli-microsoft365/cmd/aad/user/user-recyclebinitem-clear)

> - [aad user recyclebinitem list](https://pnp.github.io/cli-microsoft365/cmd/aad/user/user-recyclebinitem-list)

Reviewer 2: Seems like Patrick did a merge during the freeze period. Shall we also add the new commands here?

Author: I see... @plamber has been busy. 🤦🏼‍♂️I'll update the post Monday...

- [spo applicationcustomizer add](https://pnp.github.io/cli-microsoft365/cmd/spo/applicationcustomizer/applicationcustomizer-add)
- [spo file sharinglink clear](https://pnp.github.io/cli-microsoft365/cmd/spo/file/file-sharinglink-clear)
- [spo file sharinglink set](https://pnp.github.io/cli-microsoft365/cmd/spo/file/file-sharinglink-set)
- [spo navigation node get](https://pnp.github.io/cli-microsoft365/cmd/spo/navigation/navigation-node-get)
- [spo navigation node set](https://pnp.github.io/cli-microsoft365/cmd/spo/navigation/navigation-node-set)

## Making SharePoint Framework Development easier

Granting permissions in SharePoint Framework projects can be a bother. Developers need to deploy the package before the permissions are visible on the API access page of the SharePoint admin dashboard. This process takes a lot of small steps. In this release we introduce a new command to make this process easier. You can now grant consent to all API permissions that are included in your SPFx project. No need to execute multiple gulp commands. No need to leave VS Code. 

You can now simply run:

```sh 
m365 spfx project permissions grant
```

> ## Another steps in implementing CLI for Microsoft 365 context

Reviewer: Noticed one more typo 😄 

```suggestion
## Another step in implementing CLI for Microsoft 365 context
```

## Contributors

This release wouldn't be possible without the help of (in alphabetical order):

- [Adam Wójcik](https://github.com/Adam-it)
- [Jasey Waegebaert](https://github.com/Jwaegebaert)
- [Martin Lingstuyl](https://github.com/martinlingstuyl)
- [Mathijs Verbeeck](https://github.com/MathijsVerbeeck)
- [Milan Holemans](https://github.com/milanholemans)
- [Nanddeep Nachan](https://github.com/nanddeepn)
- [Nico De Cleyre](https://github.com/nicodecleyre)
- [Patrick Lamber](https://github.com/plamber)
- [Rabia Williams](https://github.com/rabwill)
- [Reshmee Auckloo](https://github.com/reshmee011)
- [Waldek Mastykarz](https://github.com/waldekmastykarz)

Thank you all for the time you chose to spend on CLI for Microsoft 365 and for your help to advance it!

### High fives

CLI for Microsoft 365 wouldn't be where it is today if it weren't for our users who provide us with feedback. High fives to the following people who took the time to share their feedback and ideas for improvement with us (in alphabetical order):

- [Carl Henrik Lunde](https://github.com/chlunde)
- [hsicilia](https://github.com/hsicilia)
- [Pol Van Dingenen](https://github.com/Cyhexy)

> - [Reshmee Auckloo](reshmee011)

Reviewer 2: I think we should mention people either in the contributors section or the high fives section. Since Reshmee is already in the contributors section, I suggest we remove her here.
